staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go

```diff
@@ -123,6 +128,12 @@ type ConfigFlags struct {
 	// Allows increasing qps used for discovery, this is useful
 	// in clusters with many registered resources
 	discoveryQPS float32
+
+	httpClient     *http.Client
+	httpClientLock sync.Mutex
+
+	// Affects behaviour of Exec Credentials Provider.
+	stdinInUse bool
```

Comment:
It seems like there should be a way not to have to add `stdinInUse` to the config.

What if you were to do one of these things instead:

1. Keep track of the configuration that was used when httpClient was created and if the config is different then don't reuse the existing httpClient, but create a new one instead

2. Keep a map of httpClients to cache config-specific instances of httpClient (similar to how `tlsTransportCache` works).

3. Keep two httpClient instances, one with stdin available and one without

It just seems like this PR would be a lot simpler if you can leave all the `stdinInUse` stuff inside builder.go where it is currently.

Reply:
Unfortunately there are couple of problems with these approaches. The main issue is that we are using the clients before we have parsed / recieved all the inputs for the kubectl command (please take a look at https://github.com/kubernetes/kubernetes/pull/108459#issuecomment-1058525691 for more details). And kubectl decides later if we will be using stdin or not.

1. As mentioned, the problem is that we amend the config manually later so a source of this information is needed for such tracking (like `stdinInUse`).
2. same as 1. we need to update/callback the Factory manually that we wish for a config change. It is not received externally. Also as discussed here there might be some problems with caching: https://github.com/kubernetes/kubernetes/pull/108274#discussion_r812046486
3. This could be an option but not sure how much feasible. Problem is that the caller would have to decide if it needs a client with or without stdin enabled. The caller would have to track and use this properly. I think there is bigger chance of an error and larger code maintanace with this approach.

I understand the current approach is a bit harder to grasp, but it is not such a big change (basically a callback). I have tried to document this in the code, but I welcome any additional feedback on how to improve this.